util/util.go

```diff
@@ -492,6 +492,50 @@ func RemoveHostDirectoryContent(directory string) (err error) {
 	return nil
 }
 
+func CopyForHostDirectoryContent(src, dest string) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "failed to copy the content from %v to %v for the host", src, dest)
+	}()
+
+	srcDir, err := filepath.Abs(filepath.Clean(src))
+	if err != nil {
+		return err
+	}
+	destDir, err := filepath.Abs(filepath.Clean(dest))
+	if err != nil {
+		return err
+	}
+	if strings.Count(srcDir, "/") < 2 || strings.Count(destDir, "/") < 2 {
+		return fmt.Errorf("prohibit copying the content for the top level of directory %v or %v", srcDir, destDir)
+	}
+
+	initiatorNSPath := iscsi_util.GetHostNamespacePath(HostProcPath)
+	nsExec, err := iscsi_util.NewNamespaceExecutor(initiatorNSPath)
+	if err != nil {
+		return err
+	}
+
+	// make sure srcDir and destDir are the directories rather than the file path
+	if _, err := nsExec.Execute("bash", []string{"-c", fmt.Sprintf("ls %s", filepath.Join(srcDir, "*"))}); err != nil {
+		logrus.Warnf("cannot list the content of the src directory %v for the copy: %v", srcDir, err)
+		return nil
+	}
+	if _, err := nsExec.Execute("bash", []string{"-c", fmt.Sprintf("ls %s", filepath.Join(destDir, "*"))}); err != nil {
+		logrus.Warnf("cannot list the content of the dest directory %v for the copy, will try to delete then recreate the directory: %v", destDir, err)
+		if _, err := nsExec.Execute("rm", []string{"-rf", destDir}); err != nil {
```

Comment:
This is a dangerous operation. Also, it might clean up all the existing binary engines in the directory. Why do we need to do this?

Reply:
Maybe that path is a file path rather than a directory path. And new engine images will be launched after this upgrade, then I think doing cleanup here is fine.

Reply:
I don't understand the logic here. If it's a file, then you won't need `-rf`.

Also, what if you're using existing engine images that are compatible with the current manager?